### DISCUSSION TITLE
Dynamic `pool_slots` using the XCOM value from an upstream task

### DISCUSSION THREAD

**Discussion by sid-habu** [apache/airflow#42145] [Q&A]
I have created a custom sensor and have added `pool_slots` as one of the `template_fields`. The value of pool_slots to the sensor is passed as an XCOM value from an upstream task. 

`init` of the sensor now fails with the below error as it is unable to render the template

```
File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 935, in __init__
    if self.pool_slots < 1:
       ^^^^^^^^^^^^^^^^^^^
TypeError: '<' not supported between instances of 'str' and 'int'
``` 

Q: Is there a workaround to dynamically set the `pool_slots` based on the returned value of an upstream task?

**Comment by raphaelauv** [comment]
https://github.com/apache/airflow/issues/35803

**Reply by sid-habu** [reply]
Looks like it is not possible to set the `pool` and `pool_slots` attributes dynamically as the values are fetched from the task instance before these fields are rendered.

Pooling is such a powerful feature but in our multi-tenant platform we need to control the pool and slots dynamically
